src/components/side-panel/common/dataset-title.js

```diff
@@ -117,27 +160,54 @@ export default function DatasetTitleFactory(DatasetTag) {
         showDeleteDataset,
         onTitleClick,
         removeDataset,
-        dataset
+        dataset,
+        updateDatasetColor
       } = this.props;
 
       return (
-        <StyledDatasetTitle
-          className="source-data-title"
-          clickable={Boolean(showDatasetTable || onTitleClick)}
-        >
-          <DatasetTag dataset={dataset} onClick={this._onClickTitle} />
-          {showDatasetTable ? (
-            <CenterFlexbox className="source-data-arrow">
-              <ArrowRight height="12px" />
-            </CenterFlexbox>
-          ) : null}
-          {showDatasetTable ? (
-            <ShowDataTable id={dataset.id} showDatasetTable={showDatasetTable} />
-          ) : null}
-          {showDeleteDataset ? (
-            <RemoveDataset datasetKey={dataset.id} removeDataset={removeDataset} />
-          ) : null}
-        </StyledDatasetTitle>
+        <div className="custom-palette-panel" ref={this.root}>
+          <StyledDatasetTitle
+            className="source-data-title"
+            clickable={Boolean(showDatasetTable || onTitleClick || updateDatasetColor)}
+          >
+            {updateDatasetColor ? (
```

Comment:
I would go with a different approach, I would prefer to render one `DatasetTag` and add `onClickSquare`, `tooltip` as props. You can still render Portable as part of the `DatasetTitle `

```js
<StyledDatasetTitle>
  <DatasetTag dataset={dataset} onClick={this._onClickTitle} onClickSquare={this. handleClick} tooltip={..} />
  <Portaled isOpened={this.state.displayColorPicker !== false} left={280} top={-50}>
    <CustomPicker
      color={rgbToHex(dataset.color)}
      onChange={color =>
        updateDatasetColor(dataset.id, [color.rgb.r, color.rgb.g, color.rgb.b])
      }
      onSwatchClose={this._handleClosePicker}
   />
  </Portaled>
</StyledDatasetTitle>
```